internal/resources/business_unit/resource.go

```diff
@@ -0,0 +1,934 @@
+		)
+		return
+	}
+
+	current := NewCompanyFromNative(company)
+	diags = res.State.Set(ctx, &current)
+	res.Diagnostics.Append(diags...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+}
+
+func NewCompanyResource() resource.Resource {
+	return &companyResource{}
+}
+
+var (
+	_ resource.Resource                = &divisionResource{}
+	_ resource.ResourceWithConfigure   = &divisionResource{}
+	_ resource.ResourceWithImportState = &divisionResource{}
+)
+
+type divisionResource struct {
+	client *platform.ByProjectKeyRequestBuilder
+}
+
+// Schema implements resource.Resource.
+func (b *divisionResource) Schema(_ context.Context, req resource.SchemaRequest, res *resource.SchemaResponse) {
+	res.Schema = schema.Schema{
+		Description: "Business Unit type to represent the top level of a business. Contains specific fields and values that differentiate a Company from the generic BusinessUnit.\n\n" +
+			"See also the [Business Unit API Documentation](https://docs.commercetools.com/api/projects/business-units",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Unique identifier of the Company.",
+				Computed:    true,
+			},
+			"version": schema.Int64Attribute{
+				Description: "The current version of the Company.",
+				Computed:    true,
+			},
+			"key": schema.StringAttribute{
+				Description: "User-defined unique identifier for the Company.",
+				Required:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Company.",
+				Required:    true,
+			},
+			"status": schema.StringAttribute{
+				Description: "The status of the Company.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(BusinessUnitActive, BusinessUnitInactive),
+				},
+			},
+			"associate_mode": schema.StringAttribute{
+				Description: "The association mode of the Company.",
+				Required:    true,
+			},
+			"contact_email": schema.StringAttribute{
+				Description: "The email address of the Company.",
+				Optional:    true,
+			},
+			"shipping_address_ids": schema.ListAttribute{
+				Description: "List of the shipping addresses used by the Company.",
+				Optional:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"billing_address_ids": schema.ListAttribute{
+				Description: "List of the billing addresses used by the Company.",
+				Optional:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"default_shipping_address_id": schema.StringAttribute{
+				Description: "ID of the default shipping Address.",
+				Optional:    true,
+			},
+			"default_billing_address_id": schema.StringAttribute{
+				Description: "ID of the default billing Address.",
+				Optional:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"associate": schema.ListNestedBlock{
```

Comment:
See comments on Company resource above. Maybe we can also move a lot of the shared fields here to a generic resource and extend that to not have to implement all the logic twice